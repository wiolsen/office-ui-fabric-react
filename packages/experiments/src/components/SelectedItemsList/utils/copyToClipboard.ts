export function copyToClipboard(copyText: string): void {
  const copyInput = document.createElement('input') as HTMLInputElement;
  document.body.appendChild(copyInput);

  try {
    // Try to copy the text directly to the clipboard
    copyInput.value = copyText;
    copyInput.select();
    if (!document.execCommand('copy')) {
      // The command failed. Fallback to the method below.
      throw new Error();
    }
  } catch (err) {
    // no op
  } finally {
    document.body.removeChild(copyInput);
  }
}
