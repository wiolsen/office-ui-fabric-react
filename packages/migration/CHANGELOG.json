{
  "name": "@uifabric/migration",
  "entries": [
    {
      "version": "7.0.1",
      "tag": "@uifabric/migration_v7.0.1",
      "date": "Thu, 13 Jun 2019 00:24:48 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Initial release of Fabric 7",
            "author": "Elizabeth Craig <elcraig@microsoft.com>",
            "commit": "9bc84498f2a95e5e822178214edce2a947adac38"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@uifabric/prettier-rules\" from `^7.0.0` to `^7.0.1`"
          },
          {
            "comment": "Updating dependency \"@uifabric/tslint-rules\" from `^7.0.0` to `^7.0.1`"
          }
        ]
      }
    },
    {
      "version": "0.3.0",
      "tag": "@uifabric/migration_v0.3.0",
      "date": "Wed, 12 Jun 2019 00:42:26 GMT",
      "comments": {
        "minor": [
          {
            "comment": "Non-destructive run by default, show affected files.",
            "author": "JD Huntington <jdh@microsoft.com>",
            "commit": "9dd16b5d5317ca28df279ef7c5fbfa0392370291"
          },
          {
            "comment": "Add migration for removed ColorPicker props",
            "author": "Elizabeth Craig <elcraig@microsoft.com>",
            "commit": "5f1923d49028686b99f98cbf59d04545ad85a843"
          },
          {
            "comment": "Ignore files in node_modules directories",
            "author": "JD Huntington <jdh@microsoft.com>",
            "commit": "422b15b5b28a334c93545b5717dfa47af7c44c97"
          }
        ],
        "patch": [
          {
            "comment": "ComboBox: Migration to rename deprecated props to new props.",
            "author": "JD Huntington <jdhuntington@gmail.com>",
            "commit": "403f82d28188dbe2a2ebe63a66f669f2c13caa3e"
          },
          {
            "comment": "SearchBox: Alert for onChange props arguments change.",
            "author": "JD Huntington <jdhuntington@gmail.com>",
            "commit": "403f82d28188dbe2a2ebe63a66f669f2c13caa3e"
          },
          {
            "comment": "Remove createRef imports from OUFR",
            "author": "JD Huntington <jdh@microsoft.com>",
            "commit": "56bd533c896e1df7ef755c8de64a32b1f63cde00"
          },
          {
            "comment": "Nav: import from @uifabric/legacy",
            "author": "JD Huntington <jdhuntington@gmail.com>",
            "commit": "70c46b34ff5d127c0268cd2c4eebc739e397126f"
          },
          {
            "comment": "@autobind: Migration to warn about removal.",
            "author": "JD Huntington <jdh@microsoft.com>",
            "commit": "d7dd09998172855cfd005aa2ec901a575b838619"
          },
          {
            "comment": "Remove migration for nav to legacy",
            "author": "JD Huntington <jdh@microsoft.com>",
            "commit": "1226e0857e0fc16a9884d7d011ce0e738cfeb04e"
          },
          {
            "comment": "Add migrations for removed TextField props",
            "author": "Elizabeth Craig <elcraig@microsoft.com>",
            "commit": "04983f31fedbf04c33f7cfd2395e58b5a9a7d03a"
          }
        ],
        "none": [
          {
            "comment": "Add code mod for removal of DetailsRowCheck.isSelected in favor of 'selected'",
            "author": "Kevin Coughlin <kevintcoughlin@gmail.com>",
            "commit": "1b578736ac4b80058d6b8b60d8b81f16109aed20"
          }
        ]
      }
    },
    {
      "version": "0.2.2",
      "tag": "@uifabric/migration_v0.2.2",
      "date": "Tue, 14 May 2019 07:50:30 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Update Fabric assets link",
            "author": "Peter Jahn <pejahn@microsoft.com>",
            "commit": "4cbda272f3ba109cdff14d9ff1b603564b73ea2a"
          }
        ],
        "none": [
          {
            "comment": "Run prettier on all files",
            "author": "Elizabeth Craig <ecraig12345@gmail.com>",
            "commit": "68bf892681e7cec48499ea62312aad3dee5bbfd1"
          }
        ],
        "dependency": [
          {
            "comment": "Updating dependency \"@uifabric/prettier-rules\" from `^1.0.1` to `^1.0.2`"
          },
          {
            "comment": "Updating dependency \"@uifabric/tslint-rules\" from `^1.0.1` to `^1.0.2`"
          }
        ]
      }
    },
    {
      "version": "0.2.1",
      "tag": "@uifabric/migration_v0.2.1",
      "date": "Tue, 02 Apr 2019 00:38:15 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Use ^ ranges instead of >=",
            "author": "Elizabeth Craig <ecraig12345@gmail.com>",
            "commit": "6a536804ce9cff3444e6f631925a30c58274d931"
          }
        ]
      }
    },
    {
      "version": "0.2.0",
      "tag": "@uifabric/migration_v0.2.0",
      "date": "Fri, 22 Mar 2019 17:56:40 GMT",
      "comments": {
        "minor": [
          {
            "comment": "migration: makes migration script match the name of the package",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "211e49dcf93fd79083b0c821468b26d64b24a67f"
          }
        ]
      }
    },
    {
      "version": "0.1.4",
      "tag": "@uifabric/migration_v0.1.4",
      "date": "Fri, 22 Mar 2019 12:34:41 GMT",
      "comments": {
        "patch": [
          {
            "comment": "migration: .gitignore apparently ignored all bin/ scripts and this actually makes the bin script part of the package",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "4fc3ab93baa87aecfb29952d79a07712f72b5e8d"
          }
        ]
      }
    },
    {
      "version": "0.1.3",
      "tag": "@uifabric/migration_v0.1.3",
      "date": "Thu, 21 Mar 2019 17:21:42 GMT",
      "comments": {
        "patch": [
          {
            "comment": "migration: the bin directory was being ignored, this change makes sure the npm publish step will include the bin scripts",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "9ac31fdc8a07564f15a17b63b010fa6c65986795"
          }
        ]
      }
    },
    {
      "version": "0.1.2",
      "tag": "@uifabric/migration_v0.1.2",
      "date": "Wed, 20 Mar 2019 03:15:21 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Migration: fixed the bin script's shebang to point to the right \"env\" executable",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "693183132f82cf336b4e8f8fa00f56050882dcfc"
          }
        ]
      }
    },
    {
      "version": "0.1.1",
      "tag": "@uifabric/migration_v0.1.1",
      "date": "Tue, 19 Mar 2019 18:05:11 GMT",
      "comments": {
        "patch": [
          {
            "comment": "Initial release, under construction",
            "author": "Kenneth Chau <kenotron@users.noreply.github.com>",
            "commit": "7a3bb677a144c1be006c7e7c7afe8ed9b8c353ae"
          }
        ]
      }
    }
  ]
}
