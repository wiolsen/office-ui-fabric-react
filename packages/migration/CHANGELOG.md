# Change Log - @uifabric/migration

This log was last generated on Thu, 13 Jun 2019 00:24:48 GMT and should not be manually modified.

## 7.0.1
Thu, 13 Jun 2019 00:24:48 GMT

### Patches

- Initial release of Fabric 7

## 0.3.0
Wed, 12 Jun 2019 00:42:26 GMT

### Minor changes

- Non-destructive run by default, show affected files.
- Add migration for removed ColorPicker props
- Ignore files in node_modules directories

### Patches

- ComboBox: Migration to rename deprecated props to new props.
- SearchBox: Alert for onChange props arguments change.
- Remove createRef imports from OUFR
- Nav: import from @uifabric/legacy
- @autobind: Migration to warn about removal.
- Remove migration for nav to legacy
- Add migrations for removed TextField props

## 0.2.2
Tue, 14 May 2019 07:50:30 GMT

### Patches

- Update Fabric assets link

## 0.2.1
Tue, 02 Apr 2019 00:38:15 GMT

### Patches

- Use ^ ranges instead of >=

## 0.2.0
Fri, 22 Mar 2019 17:56:40 GMT

### Minor changes

- migration: makes migration script match the name of the package

## 0.1.4
Fri, 22 Mar 2019 12:34:41 GMT

### Patches

- migration: .gitignore apparently ignored all bin/ scripts and this actually makes the bin script part of the package

## 0.1.3
Thu, 21 Mar 2019 17:21:42 GMT

### Patches

- migration: the bin directory was being ignored, this change makes sure the npm publish step will include the bin scripts

## 0.1.2
Wed, 20 Mar 2019 03:15:21 GMT

### Patches

- Migration: fixed the bin script's shebang to point to the right "env" executable

## 0.1.1
Tue, 19 Mar 2019 18:05:11 GMT

### Patches

- Initial release, under construction

