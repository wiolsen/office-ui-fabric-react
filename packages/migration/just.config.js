const { preset, just } = require('@uifabric/build');
const { task, series } = just;

preset();
// TODO remove, fix tests on windows environments
task('build', series('clean', 'copy', 'ts:commonjs-only'));
