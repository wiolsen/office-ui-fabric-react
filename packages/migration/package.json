{
  "name": "@uifabric/migration",
  "version": "7.0.1",
  "description": "Migration scripts for UI Fabric",
  "main": "lib/index.js",
  "bin": {
    "migration": "./bin/migration.js"
  },
  "typings": "lib/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/OfficeDev/office-ui-fabric-react"
  },
  "license": "MIT",
  "scripts": {
    "build": "npx just-scripts build",
    "clean": "npx just-scripts clean",
    "code-style": "npx just-scripts code-style",
    "start": "npx just-scripts dev",
    "start-test": "npx just-scripts jest-watch",
    "update-snapshots": "npx just-scripts jest -u",
    "just": "npx just-scripts"
  },
  "devDependencies": {
    "@types/jest": "23.0.0",
    "@types/glob": "^7.1.1",
    "@types/mock-fs": "3.6.30",
    "@types/node": "^8.10.29",
    "@types/semver": "^5.5.0",
    "@uifabric/prettier-rules": "^7.0.1",
    "@uifabric/tslint-rules": "^7.0.1",
    "glob": "^7.1.2",
    "mock-fs": "4.10.0",
    "@uifabric/build": "^7.0.0"
  },
  "dependencies": {
    "semver": "^5.6.0",
    "tslib": "^1.7.1",
    "riceburn": "1.3.1",
    "typescript": "3.5.1"
  }
}
