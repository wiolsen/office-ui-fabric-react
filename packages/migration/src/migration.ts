export interface IMigrationOptions {
  dryRun: boolean;
  warn: (message: string) => void;
}

// tslint:disable-next-line: interface-name
export interface ModResult {
  fileName: string;
  state: 'not-modified' | 'would-be-modified' | 'modified';
}

export interface IMigration {
  note: string;
  step: (opts: IMigrationOptions) => ModResult[];
}

const allMigrations: IMigration[] = [];

export function migration(note: string, step: (opts: IMigrationOptions) => ModResult[]): IMigration {
  return {
    note,
    step
  };
}

export function registerMigration(theMigration: IMigration) {
  allMigrations.push(theMigration);
}

export function applyRegisteredMigrations(options: IMigrationOptions) {
  console.error('Apply migration steps:');
  allMigrations.forEach(theMigration => {
    console.error(`- ${theMigration.note}`);
    const results = theMigration.step(options);
    console.log(
      results
        .filter(r => r.state === 'modified')
        .map(r => `  ${r.fileName}`)
        .join('\n')
    );
  });
}

export function warn(message: string) {
  console.error(`  ${message}`);
}
