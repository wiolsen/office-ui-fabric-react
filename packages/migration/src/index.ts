import fs from 'fs';
import path from 'path';
import semver from 'semver';
import { applyRegisteredMigrations, registerMigration, warn } from './migration';
import { CliParser, displayHelp } from './cli/cli';

const args = new CliParser().parse(process.argv.slice(2));

if (args.help) {
  displayHelp();
  process.exit(1);
}

const modsBasePath = path.join(__dirname, 'mods');

function findMigrationPaths(): string[] | null {
  if (!fs.existsSync(modsBasePath)) {
    console.log('No mods available');
    return null;
  }

  const modsPaths = fs
    .readdirSync(path.join(modsBasePath))
    .sort(semver.compare)
    .reverse();

  // let's find the biggest one that fits the range passed in from argv[2]
  const range = new semver.Range(args.version);
  const latestVersion = modsPaths.find(version => range.test(version));
  const target = latestVersion || args.version;

  return fs
    .readdirSync(path.join(modsBasePath, target))
    .filter(name => name.endsWith('.js'))
    .sort()
    .map((modPath: string) => path.join(modsBasePath, target, modPath));
}

function applyMigrations(): void {
  try {
    const modPaths = findMigrationPaths();

    if (modPaths) {
      modPaths.forEach(modPath => {
        registerMigration(require(modPath).default);
      });

      applyRegisteredMigrations({ dryRun: !args.writeResults, warn: warn });
    }
  } catch (e) {
    console.error(e);
  }
}

applyMigrations();
