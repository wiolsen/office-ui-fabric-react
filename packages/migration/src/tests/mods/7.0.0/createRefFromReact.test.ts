import migration from '../../../mods/7.0.0/createRefFromReact';
import path from 'path';
import { runMigration } from '../utils';

describe('createRefFromReact', () => {
  it('transforms createRef', async () => {
    const result = await runMigration(migration, path.join('7.0.0', 'createRef.tsx'));
    expect(result.warnings).toEqual([]);
    expect(result.contents).toMatchSnapshot();
  });
});
