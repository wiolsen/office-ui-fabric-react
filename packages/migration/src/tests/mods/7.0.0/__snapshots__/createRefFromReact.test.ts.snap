// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRefFromReact transforms createRef 1`] = `
"import { fooBarBaz, IThing } from 'office-ui-fabric-react';
import { createRef } from 'react';
import * as React from 'react';

class Foo {
  private _r = createRef<IButton>();

  constructor(props: {}) {
    super(props);
    this.state = {};
  }
}
"
`;
